django/contrib/auth/middleware.py

```diff
@@ -34,6 +38,49 @@ def process_request(self, request):
         request.auser = partial(auser, request)
 
 
+class LoginRequiredMiddleware(MiddlewareMixin):
```

Comment:
Using `MiddlewareMixin` requires a context-switch in async code. Considering that `contrib.auth` has an async interface (as applied here: you can use `await request.auser()` to get the current user, instead of `request.user`). can this middleware be rewritten to not require a context switch in async code?

Ref: https://github.com/django/django/pull/18036/commits/2c59d74e56ea6364274a5780d6d4ed32f25ce6e3

(note that that code is untested, but hopefully that demonstrates what I'm talking about)

Reply:
I've updated the above commit reference after I added testing and cleaned up the implementation.